qucumber/utils/cplx.py

```diff
@@ -172,6 +172,31 @@ def outer_prod(x, y):
     return z
 
 
+def prod(x):
+    """A function which returns the product of the complex numbers in a vector
+
+    :param x: A complex vector
+    :type x: torch.Tensor
+
+    :raises ValueError:	If x is not a complex vector with its first dimension being 2,
+                        or x is an empty vector, then an error will be raised.
+
+    :returns: The product of all of the complex numbers in x
+    :rtype: torch.Tensor
+    """
+
+    if len(list(x.size())) < 2:
+        raise ValueError("The input is not of the right dimension!")
+
+    if len(real(x)) == 0:
+        raise ("x must be non-empty!")
+
+    for i in range(len(real(x)) - 1):
+        x[:, i + 1] = scalar_mult(x[:, i], x[:, i + 1])
```

Comment:
Good thinking, I hadn't caught on to this. However, this is actually something we could do away with completely, since I forgot to remove it after I stopped calling it in the main code. However, I can fix it and keep it in, if there is some reason to?

Reply:
Removed from code as it is not used anywhere and would only cause confusion by its presence.